### PR TITLE
CI: integrate hubble-relay and add initial hubble-relay tests

### PR DESCRIPTION
**Pull request by rolinh**
This PR integrates hubble-relay into the CI and adds basic tests for hubble-relay. 

Ref: #11228

### CODE REVIEW

**Comment by Rolinh** [comment]
test-me-please

**Review by michi-covalent — APPROVED** [review]
amazing test coverage

**Review by gandro — APPROVED** [review]
Not super familiar with the provisioning scripts, but the CI tests themselves look fine.

**Comment by gandro** [comment]
test-me-please

**Comment by gandro** [comment]
retest-runtime

**Review by nebril — CHANGES_REQUESTED** [review]
Good changes! I have one small request - please add 
```
--build-arg CILIUM_SHA=$(firstword $(GIT_VERSION))
```
 to `docker build` call in hubble-relay build step in Makefile and 
```
ARG CILIUM_SHA=""
LABEL cilium-sha=${CILIUM_SHA}
```
in hubble-relay dockerfile for all intermediate images as well as the final images to ensure that these get cleaned in GKE builds.

File: hubble-relay.Dockerfile [hunk 1]
```diff
@@ -17,6 +17,8 @@ RUN go get -d github.com/google/gops && \
     strip /go/bin/gops
 
 FROM scratch
+ARG CILIUM_SHA=""
```
Comment (nebril) [inline hubble-relay.Dockerfile:26]: Please add these also to every intermediate image we are using (every `FROM` Dockerfile command creates a new image), so they need to be added on current lines 2, 8 and 11.

**Review by nebril — CHANGES_REQUESTED** [review]

**Comment by Rolinh** [comment]
test-me-please

**Comment by Rolinh** [comment]
test-me-please

**Comment by Rolinh** [comment]
retest-runtime

**Comment by Rolinh** [comment]
retest-4.9

**Review by gandro — APPROVED** [review]

**Review by nebril — APPROVED** [review]

**Comment by tklauser** [comment]
Runtime tests failed on known flake #11512 (https://jenkins.cilium.io/job/Cilium-PR-Runtime-4.9/295/), thus merging.

**Pull request merged**